src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
> it seams like you are suggesting to have a small interface in the PSR (ie only support resource) and then let someone create a utility package to make the PSR interface easy to use

That is precisely what I'm suggesting. For all the reasons you just agreed with.

> The argument I want to make is that we should be careful not to overlook the developer experience and ease of use.

Of course, what we *do* provide should be developer-friendly.

But these are *interfaces* - we're not designing a library, we're designing the abstractions that are supposed to help the libraries interoperate. If we go beyond that, we're just dictating library design - as well as adding redundancy and increasing the chance of inconsistencies between libraries.

The scope therefore must be the minimum required for interoperability. We should not have higher-order abstractions that do not help interoperability in some way - the inclusion of any higher-order abstraction not required for interoperability by definition can only be based on opinion.

Contrast this with the lowest order of abstraction - for example, there is such a thing as a `resource` in PHP, and there is such a thing as a `StreamInterface` in PSR-7, and there is objectively one way to abstract the connection between those: `createStream($resource): StreamInterface`. What you, or I, or anybody else thinks are handy or useful ways to *create* resources, or do *anything else* with resources, is completely based on opinion; it's also objectively beyond the scope of interoperability between libraries offering resource-based stream implementations, and objectively leads to duplication. It's objectively unnecessary, since we can write libraries that provide all the resource-related work anyone might want to do, *independently* of which library they choose, without duplicating any functionality, and without locking down any patterns or enforcing any aspects of design.

I'm sorry, but there is no way to design something opinionated and be objective about it.

I understand you feel differently, so I guess we'll have to agree to disagree :-)